python/coqtail.py

```diff
@@ -1292,6 +1300,24 @@ def _skip_comment(
     return _skip_block(lines, sline, scol, b"(*", b"*)")
 
 
+def _skip_elpi(
+    lines: Sequence[bytes],
+    sline: int,
+    scol: int,
+) -> Optional[Tuple[int, int]]:
+    """Skip the next block contained in lp:{{ }}."""
+    return _skip_block(lines, sline, scol, b"lp:{{", b"}}", {b"{{": _skip_br2})
```

Comment:
I tried handling `%` comments (just pushed) but it seems on Coq/elpi's side a `}}` on the right of `%` still closes the block.
Handling `%` properly (especially with nested `{{ }}` blocks and with double-quotes) is trickier than expected.

Example:

```
From elpi Require Import elpi.

Elpi Program w lp:{{
 % }} .     testplsignore
}}.
```

This gets accepted, ignoring contents after the `.` in the middle. If you remove that `.`, there is a parse error.

Reply:
I guess if Coq/Elpi isn't parsing it correctly there's not much we can do about it besides maybe open an issue. I think what you have now is correct though from Coqtail's point of view. Basically just try to treat the whole `lp:{{ .. }}` block as an opaque unit. So once you fix the tests and lints I think it's good to merge.

Reply:
I asked Enrico on Zulip and it turns out Coq just doesn't take comments into account, and it just counts nesting of braces `{{ }}` https://github.com/coq/coq/blob/f2bf445b8f4f5241ebdc348b69961041b4e57883/parsing/cLexer.ml#L542

Reply:
So I guess the question is do we have Coqtail do the same and count `{{`/`}}` even in strings/comments, or try to ignore them, give the whole block to Coq, and let it deal with it. It seems like the main difference is in the former case Coqtail might print a parse error instead of Coq.

For example
```coq
Elpi Program a lp:{{ 
coq.say "}}". 
}}. 
```

If Coqtail skips over the string and passes on the whole block Coq prints `Syntax Error: Lexer: Unterminated string`. If instead Coqtail doesn't skip the string then it prints `Found unmatched "`. 

I guess it's simplest to copy Coq's behavior and just count braces (i.e., only use `_skip_br2`). It feels pretty hacky, but I guess in practice people probably aren't using a lot of unmatched double braces in their comments or strings. Hopefully no one is putting Vim fold-marker comments in their Elpi code. Could you add a comment to `_skip_elpi` pointing to this PR to explain why it doesn't use `_skip_str` or `_skip_elpi_comment`.